docs/ha.md

```diff
@@ -0,0 +1,144 @@
+ 
+In this solution option **a** is probably best since it provides more control and the additional configuration required is minimal. Conceptually, the user might specify something like
+```
+gateway: G1
+preference: 100
+[..]
+
+gateway: G2
+preference: 20
+[..]
+
+gateway: G3
+preference: 10
+[..]
+```
+where preferences may always be distinct and higher may mean "more preferred"
+
+### solution 2
+The natural evolution of solution-1 is one where gateways are not idle; i.e. **Active-Active**. In order to meet the requirement that the traffic exchanged between any pair of vpcs goes over the same gateway in both directions, routing has to be conditioned, not on a per-VPC-basis but on a peering basis. To illustrate, suppose 3 VPCs **A**, **B** and **C** and 2 gateways **G1** and **G2** and assume that the following peerings exist
+
+```
+peering: A--B
+ A exposes [a]
+ B exposes [b]
+
+peering: A--C
+ A exposes [a]
+ C exposes [c]
+
+(*) [x] denotes a range/block/CIDR. 
+ It is irrelevant if [a], [b] and [c] are native or NATed prefixes.
+```
+
+In an active-backup setup all of the traffic between a<->b and a<->c would go over a single gateway. With this second solution, the traffic a<->B could go over GW1 and a<->c over GW2:
+
+- A leaf node where VPC A was instantiated would have a route to **b** via GW1 and one to **c** via GW2.
+- If there was a leaf where both VPCs B and C where instantiated, the leaf would have 2 routes to **a** one via **G1** and one via **G2**, in distinct VRFs.
+
+This resembles the affinity concept employed in MPLS Traffic Engineering (TE), but at the node level. In terms of configuration, the API would be augmented to indicate the preferred gateway **at the peering level**. In our example:
+```
+peering: A--B
+ A exposes [a]
+ B exposes [b]
+ preference: use primarily G1, failover to G2
+
+peering: A--C
+ A exposes [a]
+ C exposes [c]
+ preference: use primarily G2, failover to G1
+```
+The problem of this encoding is that it is tedious when N > 2 gateways are used. An alternative approach would define something like "preference sets"" or "affinity groups"" such as:
+```
+affinity-group: group-1
+ - gateway: G1
+ preference: 100
+ - gateway: G2
+ preference: 50
+ - gateway: G3,
+ preference: 20
+affinity-group: group-2
+ - gateway: G1
+ preference: 50
+ - gateway: G2
+ preference: 100
+ - gateway: G3,
+ preference: 20
+etc.
+```
+--which allows enumerating sets of gateways and ranking them-- and then "map" VPC peerings to the affinity groups as desired:
+```
+peering: A--B
+ A exposes [a]
+ B exposes [b]
+ affinity-group: group-1
+peering: A--C
+ A exposes [a]
+ C exposes [c]
+ affinity-group: group-2
+```
+**Note:**
+
+- The _preference_ within an affinity group is just one way of ranking gateways. The value itself may be irrelevant. An alternative encoding may just enumerate the gateways in ascending/descending order, like
```

Comment:
If we go for the affinity approach, I think this is preferable, as the other option is more error prone and forces us to consider the case where the user puts the same priority to different gateways in the same group